### PR TITLE
Simple Recurrent Unit

### PR DESCRIPTION
**Pull request by tjingrant**
As per https://github.com/tensorflow/tensorflow/pull/15434#issuecomment-352343576.

### CODE REVIEW

**Comment by gunan** [comment]
There are two test issues.
First with the function_test:
https://source.cloud.google.com/results/invocations/6b22c584-f73d-4c1b-b939-1c68fce02e6e/targets/%2F%2Ftensorflow%2Fpython:function_test?page=log

2nd, api_compatibility_test:
https://source.cloud.google.com/results/invocations/6b22c584-f73d-4c1b-b939-1c68fce02e6e/targets/%2F%2Ftensorflow%2Ftools%2Fapi%2Ftests:api_compatibility_test?page=log
```
    $ bazel build tensorflow/tools/api/tests:api_compatibility_test
    $ bazel-bin/tensorflow/tools/api/tests/api_compatibility_test \
          --update_goldens True

You will need an API approval to make changes to the public TensorFlow API. This
includes additions to the API.
```
These instructions should be followed on a linux machine with python 2.

**Comment by tjingrant** [comment]
@gunan the function test seems to be unrelated to this PR. Specifically, the error msg was:
```
No gradient defined for op: ClipByValue
```

**Comment by yifeif** [comment]
@tensorflow-jenkins test this please

**Comment by ebrevdo** [comment]
A better solution would be to move the SRUCell into tf.contrib.rnn python/ops (rnn_cell.py maybe?) since this was never meant to go directly into core.

**Comment by tjingrant** [comment]
@ebrevdo sounds good, I'll revert my last commit and instead move it under contrib.

File: tensorflow/python/ops/rnn_cell_impl.py [hunk 1]
```diff
@@ -289,6 +288,86 @@ def __call__(self, inputs, state, scope=None):
     # method.  See the class docstring for more details.
     return base_layer.Layer.__call__(self, inputs, state, scope=scope)
 
+class SRUCell(_LayerRNNCell):
+  """Training RNNs as Fast as CNNs (cf. https://arxiv.org/abs/1709.02755).
```
Comment (asimshankar) [inline tensorflow/python/ops/rnn_cell_impl.py]: Drive-by comment: Can you expand on the description here? This seems a bit terse.

Things to include:
- What does "SRU" stand for :)
- A description of the basic characteristics of this cell.

**Comment by tjingrant** [comment]
Hi @ebrevdo should I stick to `_LayerRNNCell ` as the base class or switch back to `RNNCell` as the base class as for other rnn architectures in rnn_cell.py in the contrib directory (e.g., https://github.com/tensorflow/tensorflow/blob/810394550571c5feb333cb6da66afb4b20c3bd85/tensorflow/contrib/rnn/python/ops/rnn_cell.py#L92).

**Comment by ebrevdo** [comment]
Use the _LayerRNNCell; the other contrib classes will be moved over to that
eventually as well.

**Comment by tjingrant** [comment]
@ebrevdo @asimshankar should be good now.

**Review by ebrevdo — APPROVED** [review]

**Comment by yifeif** [comment]
@tensorflow-jenkins test this please
